packages/components/fields/search-select-field/src/search-select-field.tsx

```diff
@@ -1,320 +1,380 @@
 import FieldErrors from '@commercetools-uikit/field-errors';
 
-const hasErrors = (errors) => errors && Object.values(errors).some(Boolean);
-const sequentialId = createSequentialId('async-select-field-');
+type ReactSelectAsyncProps = AsyncProps<unknown, boolean, GroupBase<unknown>>;
 
-const SearchSelectField = (props) => {
-  const hasError = Boolean(props.touched) && hasErrors(props.errors);
-  const id = props.id || sequentialId();
-  return (
-    <Constraints.Horizontal max={props.horizontalConstraint}>
-      <Spacings.Stack scale="xs">
-        <FieldLabel
-          title={props.title}
-          hint={props.hint}
-          description={props.description}
-          onInfoButtonClick={props.onInfoButtonClick}
-          hintIcon={props.hintIcon}
-          badge={props.badge}
-          hasRequiredIndicator={props.isRequired}
-          htmlFor={id}
-        />
-        <SearchSelectInput
-          {...filterDataAttributes(props)}
-          horizontalConstraint="scale"
-          hasError={hasError}
-          aria-label={props['aria-label']}
-          aria-labelledby={props['aria-labelledby']}
-          isAutofocussed={props.isAutofocussed}
-          backspaceRemovesValue={props.backspaceRemovesValue}
-          components={props.components}
-          filterOption={props.filterOption}
-          id={id}
-          containerId={props.containerId}
-          isClearable={props.isClearable}
-          isDisabled={props.isDisabled}
-          isReadOnly={props.isReadOnly}
-          isOptionDisabled={props.isOptionDisabled}
-          isMulti={props.isMulti}
-          hasWarning={props.hasWarning}
-          maxMenuHeight={props.maxMenuHeight}
-          menuPortalTarget={props.menuPortalTarget}
-          menuPortalZIndex={props.menuPortalZIndex}
-          menuShouldBlockScroll={props.menuShouldBlockScroll}
-          name={props.name}
-          noOptionsMessage={props.noOptionsMessage}
-          loadingMessage={props.loadingMessage}
-          onBlur={props.onBlur}
-          onChange={props.onChange}
-          onFocus={props.onFocus}
-          onInputChange={props.onInputChange}
-          placeholder={props.placeholder}
-          tabIndex={props.tabIndex}
-          tabSelectsValue={props.tabSelectsValue}
-          value={props.value}
-          // Async react-select props
-          loadOptions={props.loadOptions}
-          cacheOptions={props.cacheOptions}
-          showOptionGroupDivider={props.showOptionGroupDivider}
-          optionType={props.optionType}
-        />
-        <FieldErrors
-          errors={props.errors}
-          isVisible={hasError}
-          renderError={props.renderError}
-        />
-      </Spacings.Stack>
-    </Constraints.Horizontal>
-  );
+type TEvent = {
+  target: {
+    name?: string;
+    value?: unknown;
+  };
+  persist: () => void;
 };
 
-SearchSelectField.displayName = 'SearchSelectField';
-SearchSelectField.propTypes = {
+type TFieldError = Record<string, boolean>;
```

Comment:
```suggestion
type TFieldErrors = Record<string, boolean>;
```

I would use plural here because this could be an object with several errors (key/value pairs)